anyway/views/news_flash/api.py

```diff
@@ -292,12 +293,45 @@ def normalize_query(params: dict):
     return {k: normalize_query_param(k, v) for k, v in params_non_flat.items()}
 
 
+def update_location_verification_history(
+    user_id: int,
+    news_flash_id: int,
+    prev_location: str,
+    prev_qualification: int,
+    new_location: str,
+    new_qualification: int,
+):
+    new_location_qualifiction_history = LocationVerificationHistory(
+        user_id=user_id,
+        news_flash_id=news_flash_id,
+        location_verification_before_change=prev_qualification,
+        location_before_change=prev_location,
+        location_verification_after_change=new_qualification,
+        location_after_change=new_location,
+    )
+    db.session.add(new_location_qualifiction_history)
+    db.session.commit()
+
+
+def extracted_location_and_qualifiction(news_flash_obj: NewsFlash):
+    road_segment_name = news_flash_obj.road_segment_name
+    yishuv_name = news_flash_obj.yishuv_name
+    street1_hebrew = news_flash_obj.street1_hebrew
+    if road_segment_name is not None:
+        location = json.dumps({"road_segment_name": road_segment_name})
+    elif (yishuv_name is not None) and (street1_hebrew is not None):
+        location = json.dumps({"yishuv_name": yishuv_name, "street1_hebrew": street1_hebrew})
```

Comment:
We will be working with additional resolutions soon. I prefer that using the `resolution_dict` (defined in `anyway/parsers/__init__.py`), the code will not apply logic on specific field names, but use the list of fields from `resolution_dict`.

Reply:
done, the only issue is what to do if the field is None, should i keep him in the db?